core/src/autogluon/core/models/abstract/abstract_model.py

```diff
@@ -539,61 +539,73 @@ def fit_with_prune(self, X, y, X_val, y_val, max_num_fit=3, prune_threshold=0.,
+                    X_fi, y_fi = X_val, y_val
+                fitted_copies_info.append((round(score, 4), self_copy.path))
                 if best_score is None or score >= best_score:
-                    logger.log(20, f"\tFit {index+1}: Current score {score} is better than best score {best_score}.")
+                    if best_score is None:
+                        logger.log(20, f"\tFit {index+1}: Current score is {score}")
+                    else:
+                        logger.log(20, f"\tFit {index+1}: Current score {score} is not worse than best score {best_score}.")
                     best_score = score
                     best_index = index
-                if index - best_index == stop_threshold:
+                if index - best_index >= stop_threshold:
                     logger.log(20, f"\tEnding prune loop after {index+1} iterations. Best score: {best_score}")
                     break
 
                 # compute all feature importance and remove ones that don't meet the threshold
-                importance_df = self_copy.compute_feature_importance(X=X_val, y=y_val, num_shuffle_sets=num_shuffle_sets)
+                importance_df = self_copy.compute_feature_importance(X=X_fi, y=y_fi, num_shuffle_sets=num_shuffle_sets)
                 cols_to_drop_df = importance_df[importance_df['importance'] <= prune_threshold]
                 cols_to_drop = list(cols_to_drop_df.index)
-                cols_to_drop_importance = list(cols_to_drop_df['importance'])
+                cols_to_drop_importance = list(map(lambda importance: round(importance, 4), cols_to_drop_df['importance']))
                 if 0 < len(cols_to_drop) < len(self_copy.features):
                     logger.log(20, f"\tWill try fit after pruning these columns that failed to meet importance threshold {prune_threshold}:")
                     logger.log(20, f"\t{list(zip(cols_to_drop, cols_to_drop_importance))}")
                     current_features = [feat for feat in self_copy.features if feat not in cols_to_drop]
                 else:
+                    logger.log(20, f"\tNo columns to drop. Ending prune loop after {index+1} iterations. Best score: {best_score}")
                     break
         finally:
             # We can alternatively return a list of fitted models
             if len(fitted_copies_info) > 0:
                 best_model_path = fitted_copies_info[best_index][1]
-                best_model = self.load(best_model_path)
-                # original_model_path = self.path
-                # self.__dict__.update(best_model.__dict__)
-                # self.path = original_model_path
+                best_model = self.__class__.load(best_model_path)
             else:
-                best_model = self
-            # Cleanup saved models
-            for info in fitted_copies_info:
-                tmp_model = AbstractModel.load(info[1])
+                best_model = self_copy
```

Comment:
When this is returned, what is going to happen?

Reply:
I had that there in case `fit` errors out at the very end or `score` errors out. If the model can still generate predictions, a trained model object will be returned. Otherwise it will return an incomplete model.

Reply:
Do we want an incomplete model to be returned?